### PR TITLE
Powerbash10k theme

### PR DESCRIPTION
**Pull request by awerebea**
Simple colorful terminal prompt theme (inspired by [powerlevel10k](https://github.com/romkatv/powerlevel10k) zsh theme, based on [Brainy](https://github.com/ohmybash/oh-my-bash/tree/master/themes/brainy) theme).

### CODE REVIEW

**Review by akinomyoga — CHANGES_REQUESTED** [review]
Thank you for your contribution.

- Could you add the entry in `themes/THEMES.md`?
- Could you resize the image to have an approximate size of `(600-800)x290`? The current size 2467x329 is too large in the horizontal direction.

**Comment by akinomyoga** [comment]
Thank you. The permission of the image file seems to be still not fixed.

```bash
$ ls -l powerbash10k-dark.png
-rwxr-xr-x 1 murase murase 104871 2023-09-15 04:56:06 powerbash10k-dark.png
```

If you don't know how to change the permission, you can search for the command `chmod`.

**Comment by awerebea** [comment]
Oh, I know the `chmod` command, I just understand yours:

> Also, the resolution of the image file is incorrect. The image file is not an executable file.

Like "...isn't an executable, but it should be." 😆

My bad, sorry, it's already fixed.

File: themes/powerbash10k/powerbash10k.theme.sh [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+}
+
+function __pb10k_ps2 {
+    local color
+    color=$_omb_prompt_bold_white
+    printf "%s%s%s" "$color" "${__PB10K_PROMPT_CHAR_PS2}  " "$_omb_prompt_normal"
+}
+
+function _omb_theme_PROMPT_COMMAND {
+    exitcode="$?"
```
Comment (akinomyoga) [inline themes/powerbash10k/powerbash10k.theme.sh:361]:
```suggestion
    local exitcode="$?"
```

Note: Bash's variable scoping model is dynamic scoping, i.e., the variables declared (as `local`) at the caller function are visible to the callee functions (in the entire call tree).

File: themes/powerbash10k/powerbash10k.theme.sh [hunk 2]
```diff
@@ -0,0 +1,367 @@
+#! bash oh-my-bash.module
+
+# powerbash10k theme for oh-my-bash, inspired by powerlevel10k zsh theme
+# made by awerebea, based on brainy theme by MunifTanjim
+
+# Helpers
+function __pb10k_remove_empty_elements {
+    IFS="|" read -ra origin_array <<< "$1"
```
Comment (akinomyoga) [inline themes/powerbash10k/powerbash10k.theme.sh:8]: I'm not sure if it is intensional or not, but why doesn't this function have declarations of (seemingly) local variables despite that all the other functions seem to carefully declare local variables? Are these variables used outside this function? Maybe, can we make these variables local?

```suggestion
    local origin_array new_array element trimmed
    IFS="|" read -ra origin_array <<< "$1"
```

File: themes/powerbash10k/powerbash10k.theme.sh [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+    __BOTTOM+="${args[0]}${args[1]}"
+    [ ${#args[1]} -gt 0 ] && __BOTTOM+=" "
+}
+
+function __pb10k_top {
+    local seg segments info terminal_width filler_character
+    __TOP_LEFT=""
+    __TOP_RIGHT=""
+    __TOP_RIGHT_LEN=0
+    __SEG_AT_RIGHT=0
```
Comment (akinomyoga) [inline themes/powerbash10k/powerbash10k.theme.sh:89]: Could these variables also be localized?

File: themes/powerbash10k/powerbash10k.theme.sh [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+    segments="${*:-}"
+    typeset func
+    case "$action" in
+        show)
+        func=__pb10k_show;;
+        hide)
+        func=__pb10k_hide;;
+    esac
+    IFS=" " read -ra segments <<< "$segments"
+    for seg in "${segments[@]}"; do
```
Comment (akinomyoga) [inline themes/powerbash10k/powerbash10k.theme.sh:307]:
```suggestion
    local seg
    for seg in "${segments[@]}"; do
```

**Comment by akinomyoga** [comment]
> My bad, sorry, it's already fixed.

Thanks!

File: themes/powerbash10k/powerbash10k.theme.sh (hunk 4 above)
Comment (awerebea) [reply themes/powerbash10k/powerbash10k.theme.sh:307]: Sure, I'll localize all these variables.

Comment (awerebea) [reply themes/powerbash10k/powerbash10k.theme.sh:308]: Fixed in https://github.com/ohmybash/oh-my-bash/pull/474/commits/043379ae88aabd7d636999feaf454b289abcc2e5.

**Review by akinomyoga — APPROVED** [review]

**Comment by akinomyoga** [comment]
Thank you for updating the PR! I've merged it.

**Pull request merged**